modules/drivers/redshift/src/metabase/driver/redshift.clj

```diff
@@ -196,76 +196,60 @@
-  ;; redshift doesn't have `age`, so we have to use `extract` instead.
   [driver [_ x y unit]]
-  (let [x (hx/->timestamp (sql.qp/->honeysql driver x))
-        y (hx/->timestamp (sql.qp/->honeysql driver y))]
-    (case unit
-      :year
-      (let [positive-diff (fn [a b] ; precondition: a <= b
-                            (hx/-
-                             (hx/- (extract :year b) (extract :year a))
-                             ;; decrement if a is later than b in the year calendar
-                             (hx/cast
-                              :integer
-                              (hsql/call
-                               :or
-                               (hsql/call :> (extract :month a) (extract :month b))
-                               (hsql/call
-                                :and
-                                (hsql/call := (extract :month a) (extract :month b))
-                                (hsql/call :> (extract :day a) (extract :day b)))))))]
-        (hsql/call :case (hsql/call :<= x y) (positive-diff x y) :else (hx/* -1 (positive-diff y x))))
-
-      :quarter
-      (let [positive-diff
-            (fn [a b] ; precondition: a <= b
-              (hx/cast :integer
-                       (hx/floor
-                        (hx// (hx/- (hsql/call :datediff (hsql/raw "month") a b)
-                                    (hx/cast :integer
-                                             ;; coalesce is needed because extract(day, a) > extract(day, b)
-                                             ;; returns null if a and b are equal
-                                             (hsql/call :coalesce
-                                                        (hsql/call :> (extract :day a) (extract :day b))
-                                                        false)))
-                              3))))]
-        (hsql/call :case (hsql/call :<= x y) (positive-diff x y) :else (hx/* -1 (positive-diff y x))))
-
-      :month
-      (let [positive-diff
-            (fn [a b] ; precondition: a <= b
-              (hx/-
-               (hsql/call :datediff (hsql/raw "month") a b)
-               (hx/cast :integer
-                        ;; coalesce is needed because extract(day, a) > extract(day, b)
-                        ;; returns null if a and b are equal
-                        (hsql/call :coalesce
-                                   (hsql/call :> (extract :day a) (extract :day b))
-                                   false))))]
-          (hsql/call :case (hsql/call :<= x y) (positive-diff x y) :else (hx/* -1 (positive-diff y x))))
-
-      :week
-      (let [positive-diff (fn [a b]
-                            (hx/cast
-                             :integer
-                             (hx/floor
-                              (hx// (hsql/call :datediff (hsql/raw "day") a b) 7))))]
-        (hsql/call :case (hsql/call :<= x y) (positive-diff x y) :else (hx/* -1 (positive-diff y x))))
```

Comment:
These uses of `positive-diff` were here because I thought this was required to round up when the diff was negative and round down when positive. But integer division does this anyway in redshift.